### ISSUE TITLE
AssignmentInConditionSniff throws error when encountering unclosed block

### ISSUE THREAD

**Issue by carloscz** [slevomat/coding-standard#1420]
Hi, sniff AssignmentInConditionSniff doesn't work with unclosed blocks ( tested with do/while ) and throws Undefined array key "scope_closer" error.

sample code:
```php
<?php

do
    echo "";
while (true);
```

error:

` 1 | ERROR | [ ] An error occurred during processing; checking has been aborted. The error message was: Undefined array key "scope_closer" in .../vendor/slevomat/coding-standard/SlevomatCodingStandard/Sniffs/ControlStructures/AssignmentInConditionSniff.php on line 47`

**Comment by kukulich** [comment]
Yes, we don't support this.

**Comment by carloscz** [comment]
I know it is bad coding practice but I would assume different sniff should warn about that and this one should run correctly :) but thanks for Your reply nevertheless...
